ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useRef, useMemo, useEffect } from 'react';
+import Input, { InputProps } from './Input';
+import Modal from '../Modal';
+import cx from 'classnames';
+import { StatusWithMessage } from './StatusMessage';
+
+type ToleranceRange = { label: string; value: number };
+
+export type ToleranceValues = {
+  minusTolerance: number;
+  plusTolerance: number;
+};
+
+export type TolerancePickerProps = Omit<InputProps, 'value'> & {
+  onToleranceChange: (toleranceValues: ToleranceValues) => void;
+  toleranceRanges: ToleranceRange[];
+  toleranceValues?: ToleranceValues;
+};
+
+const TolerancePicker: React.FC<TolerancePickerProps> = ({
+  onToleranceChange,
+  toleranceRanges,
+  toleranceValues: { minusTolerance, plusTolerance } = {
+    minusTolerance: toleranceRanges[0].value,
+    plusTolerance: toleranceRanges[0].value,
+  },
+  ...inputProps
+}) => {
+  const formatToleranceValue = (minusIndex: number, plusIndex: number) =>
+    `-${toleranceRanges[minusIndex]?.label || ''}/+${toleranceRanges[plusIndex]?.label || ''}`;
+
+  const [showPicker, setShowPicker] = useState(false);
+  const [inputValue, setInputValue] = useState(formatToleranceValue(0, 0));
+  const [warningStatus, setWarningStatus] = useState<StatusWithMessage | undefined>(undefined);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const { minusToleranceIndex, plusToleranceIndex } = useMemo(() => {
+    const minusToleranceIndex = toleranceRanges.findIndex(
+      (range) => range.value === minusTolerance
+    );
+    const plusToleranceIndex = toleranceRanges.findIndex((range) => range.value === plusTolerance);
+    if (minusToleranceIndex < 0 || plusToleranceIndex < 0)
+      setWarningStatus({
+        status: 'warning',
+        message: "One of the values isn't included in the ranges provided.",
```

Comment:
Maybe this message is more explicit ?
```suggestion
        `${minusToleranceIndex === -1 ? minusTolerance : plusTolerance} is not a valid tolerance value.`
```

Reply:
Ah I don't know how it was done in the speedSpaceChart. Here it would be better to have a function which take the invalid tolerance and return the translated string.
In osrd, we can give this function as a prop:
```
translation={(invalidTolerance) => t('toleranceIsInvalid', { invalidTolerance })}
```
where the translation 'toleranceIsInvalid' is translated in our json files, what do you think ?